kernel-rpm-script

```diff
@@ -0,0 +1,302 @@
+	if [ "$basepackage" = 1 -a "$YAST_IS_RUNNING" != "" ]; then
+	    mydf="$( POSIXLY_CORRECT=1 df -P /boot/ | awk '/^(\/|-[[:blank:]])/{ print $4}' )"
+	    if test "$mydf" != "" ; then
+		echo "Free diskspace below /boot: $mydf blocks"
+		# echo "512 byte blocks: $(( 2 * 1024 * 20 ))"
+		if test "$mydf" -lt  "40960" ; then
+		    echo "make room for new kernel '"$flavor"' because there are less than 20MB available."
+		    # disabled because it breaks patch rpms
+		    #rm -fv /boot/"$image"-*-"$flavor"
+		    rm -fv /boot/initrd-*-"$flavor"
+		fi
+	    fi
+	fi
+
+
+	# On AArch64 we switched from 64k PAGE_SIZE to 4k PAGE_SIZE. Unfortunately
+	# btrfs can only use file systems created with the same PAGE_SIZE. So we
+	# check if the user has any btrfs file systems mounted and refuse to install
+	# in that case.
+	if [ $( uname -m ) = aarch64 -a \
+	    "$( zgrep CONFIG_ARM64_64K_PAGES=y /proc/config.gz )" -a \
+	    "$flavor" = default ]; then
+	if [ "$FORCE_4K" = 1 ]; then
+	    # The user knows what he's doing, let him be.
+	    exit 0
+	fi
+
+	if [ "$YAST_IS_RUNNING" = "instsys" ]; then
+	    # We're probably test installing the kernel, that should succeed
+	    exit 0
+	fi
+
+	cat >&2 <<-EOF
+
+	You are running on a 64kb PAGE_SIZE kernel. The default kernel
+	switched to 4kb PAGE_SIZE which will prevent it from mounting btrfs
+	or the swap partition.
+
+	To ensure that your system still works, I am refusing to install
+	this kernel. If you want to force installation regardlesss, reinstall
+	with the environment variable FORCE_4K set to 1.
+
+	To stay with a 64kb PAGE_SIZE kernel, please follow these steps:
+
+	$ zypper in kernel-64kb
+	[ reboot into the new kernel ]
+	$ zypper rm kernel-default
+
+	You will then be on the 64kb PAGE_SIZE kernel and can update your
+	system normally.
+	EOF
+
+	exit 1
+    fi
+    [ -z "$certs" ] || [ "$basepackage" -ne 1 ] || /usr/lib/suse-module-tools/kernel-cert-$op --certs "$certs" "$@"
+
+    ;;
+preun)
+    if [ "$basepackage" -ne 0 ]; then
+	rpm -ql $nvr | grep '\.ko\(\.xz\|\.gz\|\.zst\)\?$' > /var/run/rpm-$nvr-modules
+    fi
+
+    [ -z "$certs" ] || [ "$basepackage" -ne 1 ] || /usr/lib/suse-module-tools/kernel-cert-$op --certs "$certs" "$@"
+
+    ;;
+post)
+    # Flag to trigger /etc/init.d/purge-kernels on next reboot (fate#312018)
+    # ... but avoid the first installion (bsc#1180058)
+    test $1 -gt 1 && trigger_purge_kernels || script_rc=$?
+
+    for x in /boot/"$image" /boot/initrd; do
+	rm -f $x
+	ln -s ${x##*/}-"$kernelrelease"-"$flavor" $x
+    done
+
+    if [ "$usrmerged" -ne 0 ] ; then
+	# compat stuff for /boot.
+	# if /boot and /usr are not speparate partitions we can just link
+	# the kernel there to save space. Otherwise copy.
+	if mountpoint -q /boot || mountpoint -q /usr; then
+	    copy_or_link="cp -a"
+	else
+	    copy_or_link="ln -sf"
+	fi
+	# XXX: need to fix suse-module-tools for sysctl.conf and System.map
+	for x in "$image" sysctl.conf System.map; do
```

Comment:
see bug 1189879, we need a fix here. Should be done separately though (probably before this is moved into **s-m-t**.

Reply:
Sounds like this is separate, orthogonal issue.

Reply:
right, this comment is only about the ordering of fixes.